### PR TITLE
layers: Fix plane compatibility validation

### PR DESCRIPTION
**Pull request by gnl21**
The plane index was being confused with the aspect mask which meant that
the check was failing incorrectly.

### CODE REVIEW

**Comment by gnl21** [comment]
@tobine, I wasn't thinking about those VUs, but rather about whether my change was breaking the expected error checking for this (although it was already not working). The old code was deliberately setting the 'plane' field to an invalid value if the aspect_mask didn't contain one of the PLANE_x aspects and by passing the aspect_mask directly to the other function that step has been skipped. That gives me the feeling that I've removed an error check (although the invalid value would have been ignored by the subsequent function), but I'm not sure what the correct place to make that work would be.

As an example of how this format check might go wrong, consider creating a view of G8_B8_R8_3PLANE using aspect mask COLOR_BIT and format R8_UNORM. The format check should fail since, if I'm using COLOR_BIT, I should be using the whole multi-plane format. At present the code will pass the aspect mask to FindMultiplaneCompatibleFormat which will call GetPlaneIndex(aspect_mask). That function returns 0 on error, which is a valid plane index, so the format of plane 0 (R8_UNORM) will be returned as the compatible format and the format check will pass incorrectly. It's possible that something has already checked that the aspect mask is valid but I find the behaviour a bit hard to follow in error cases like this where valid data ends up being generated from invalid inputs.

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -4120,24 +4120,24 @@ bool CoreChecks::PreCallValidateCreateImageView(VkDevice device, const VkImageVi
...
-                    default:
-                        break;
-                }
-
-                VkFormat compat_format = FindMultiplaneCompatibleFormat(image_format, plane);
+                VkFormat compat_format = FindMultiplaneCompatibleFormat(image_format, aspect_mask);
                 if (view_format != compat_format) {
+                    // View format must match the multiplane compatible format
+                    uint32_t plane = 3;  // invalid
+                    switch (aspect_mask) {
```
Comment (daveh-lunarg) [inline layers/buffer_validation.cpp]: Slight preference to export the GetPlaneIndex() from vk_format_utils and use it here, in place of the case statement.  It has no range checking (as mentioned in review), but not needed here I think.

Comment (gnl21) [reply layers/buffer_validation.cpp]: I've added a commit that changes this to use the helper function.

Comment (gnl21) [reply layers/buffer_validation.cpp]: Looks like I tested the wrong version though. I'll fix it in a moment.

**Comment by daveh-lunarg** [comment]
By inspection this fixes the bug, and passes local and github CI. I'll merge.  Thanks for the PR, @gnl21 !
